articles/how-to-execute-linux-commands-in-python/index.md

```diff
@@ -0,0 +1,91 @@
+
+**Linux** is one of the most popular operating systems used by software developers and system administrators. It is open-source, free, customizable, and is very robust and adaptable making it an ideal choice for servers, [Virtual machines(VMs)](https://www.vmware.com/topics/glossary/content/virtual-machine), and an array of other use cases. Therefore, it is essential for anyone working in the tech industry to know how to work with Linux because it is used almost everywhere. *In this tutorial, we are going to look at how we can automate and run Linux commands in Python*. 
+
+### Table of contents
+[Prerequisites](#prerequisites)
+[Introduction](#introduction)
+[Building an application to ping servers](#building-an-application-to-ping-servers)
+[Code](#code)
+[Conclusion](#conclusion)
+
+### Prerequisites
+1. Basic understanding of Linux: https://www.section.io/engineering-education/what-is-linux/ and https://www.section.io/engineering-education/introduction-to-shell-scripting/
+2. Basic programming skills in Python: https://www.python.org/about/gettingstarted/
+
+### Introduction
+Python has a rich set of libraries that allow us to execute shell commands. A naive approach would be to use the `os` library:
+
+```python
+import os
+cmd = 'ls -l'
+os.system(cmd)
+```
+The `os.system()` function allows users to execute commands in Python.  The above program lists all the files inside a directory. However, we can't read and parse the output of the command. In some commands, it is imperative to read the output and analyze it. The `subprocess` library provides a better, safer, and faster approach for this and allows us to view and parse the output of the commands.  
```

Comment:
> The `subprocess` library provides a better, safer, and faster approach for this and allows us to view and parse the output of the commands.  

Why is `subprocess` safer and faster than `os`?
It would be great if you could compare and contrast the differences and conclude how `subprocess` is better. This would let the readers understand in-depth why `subprocess` is a better library.